src/Microsoft.TemplateEngine.Cli/PostActionProcessors/AddReferencePostActionProcessor.cs

```diff
@@ -102,7 +102,26 @@ private bool AddReference(IEngineEnvironmentSettings environment, IPostAction ac
                     Reporter.Error.WriteLine(LocalizableStrings.Generic_NoCallbackError);
                     return false;
                 }
+
+                // replace the path to the project file in case it has been renamed
+                Glob g = Glob.Parse(referenceToAdd);
+
+                if (creationEffects.FileChanges != null)
+                {
+                    foreach (IFileChange2 change in creationEffects.FileChanges)
+                    {
+                        if (g.IsMatch(change.SourceRelativePath))
+                        {
+                            referenceToAdd = Path.GetFullPath(change.TargetRelativePath, outputBasePath);
+
+                            break;
+                        }
+                    }
+                }
+
+                // retrieve the full path in case the file has not been renamed
                 referenceToAdd = Path.GetFullPath(referenceToAdd, outputBasePath);
```

Comment:
Is the double execution of `Path.GetFullPath` intentional in case there is a filename change?

Seems like the above assignment of `referenceToAdd` should be just:

```
referenceToAdd = referenceNameChange.TargetRelativePath;
```

So that prepending outputBasePath happens just once